spec/rubocop/cop/lint/interpolation_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Lint::InterpolationCheck do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense for interpolation in single quoted string' do
+    inspect_source(cop, <<-END.strip_indent)
+      'asdfasfasfds\#{testvar}alkdjfhaslkjhdfadslkjf'
+    END
+    expect(cop.offenses.size).to eq(1)
+  end
+
+  it 'does not register an offense for properly interpolation strings' do
+    expect_no_offenses(<<-END.strip_indent)
+      hello = "asdfasfasfds\#{testvar}alkdjfhaslkjhdfadslkjf"
```

Comment:
Also if you put single quotes around the heredoc then you don't need to escape the interpolation. Like `<<-'END'`